### PR TITLE
Ortmodule override torch.manual_seed()

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Override torch.manual_seed() with onnxruntime.set_seed()

### CODE REVIEW

File: orttraining/orttraining/python/training/ortmodule/__init__.py [hunk 1]
```diff
@@ -58,5 +58,20 @@
 from ._custom_autograd_function import enable_custom_autograd_support
 enable_custom_autograd_support()
 
+# Override torch.manual_seed and torch.cuda.manual_seed
+def override_torch_manual_seed(seed):
+    from onnxruntime import set_seed
+    set_seed(seed)
+    return torch_manual_seed(seed)
+torch_manual_seed = torch.manual_seed
+torch.manual_seed = override_torch_manual_seed
```
Comment (baijumeswani) [inline orttraining/orttraining/python/training/ortmodule/__init__.py:67]: What happens if the user sets the seed on the PyTorch model before importing ORTModule? In that case, we would not be able to align with the set seed, right?

I was looking at the PyTorch docs and came across torch.get_rng_state and torch.set_range_state (https://pytorch.org/docs/stable/torch.html#torch.get_rng_state). Do you think we could leverage that inside the ORTModule constructor by setting the onnxruntime seed with the help of a byte tensor?

**Comment by thiagocrepaldi** [comment]
@SherlockNoMad could you set context on why this is need, what problem are you solving, why calling pytorch set seed and ort set seed separately is an issue, etc?

It seems this PR parts scenarios in two. 1) setting seed after ORTModule is imported/instantiated and 2) setting seed before that. We would have two different behaviors

**Review by thiagocrepaldi — DISMISSED** [review]

File: orttraining/orttraining/python/training/ortmodule/__init__.py (hunk 1 above)
Comment (SherlockNoMad) [reply orttraining/orttraining/python/training/ortmodule/__init__.py:67]: addressed. 
Thank you for pointing out that torch.initial_seed() is the way to read the seed.

**Review by baijumeswani — APPROVED** [review]

**Review by thiagocrepaldi — APPROVED** [review]
